tests/conftest.py

```diff
@@ -162,6 +164,14 @@ def _build_migrations_cache() -> None:
                     f"SELECT name, create_table_query FROM system.tables WHERE database='{database}'"
                 )
                 for table_name, create_table_query in rows.results:
+                    if "MATERIALIZED VIEW" in create_table_query:
+                        mv_tables.append((table_name, create_table_query))
+                    else:
+                        non_mv_tables.append((table_name, create_table_query))
+                # when using the MIGRATIONS_CACHE we should make sure local
+                # tables are created before materialized views that depend on them
+                all_tables = mv_tables + non_mv_tables
```

Comment:
I don't know why this hasn't caused a problem until now tbh since I kept getting `DB::Exception: Table snuba_test.generic_metric_counters_raw_local doesn't exist. Stack trace:` when running this for 22.8 locally too - but the other versions, including 22.3 seemed to work? anyhoo I think is just a weirdness of our tests cause running all the tests for `test_runner` individually work fine.

Reply:
Interesting. Maybe old versions of clickhouse maintained the order of creation in the query result.

Reply:
yeah perhaps